docs/architecture/adr-009-non-interactive-default-rules-for-reduced-padding.md

```diff
@@ -0,0 +1,177 @@
+
+If you choose k²/4 as the worst constructible message it would still have O(sqrt(n)) subtree roots. This is because the size of the message is k²/4 with a width of k and a length of k/4. This means the number of rows the message fills approaches O(sqrt(n)). Therefore we need to find a message where the number of rows is log(n) of the size of the message.
+
+With k being the square size and n being the number of shares and r being the number of rows, we want to find a message so that:
+k * r = n & log(n) = r => k = n/log(n)
+
+By substituting in k we can calculate n. To get r we calculate n/k, rounding up to the next highest integer in the processes.
+
+|   k  |   n   |  r |
+|:----:|:-----:|:--:|
+|    2 |     4 |  2 |
+|    4 |    16 |  4 |
+|    8 |    43 |  6 |
+|   16 |   108 |  7 |
+|   32 |   256 |  8 |
+|   64 |   589 | 10 |
+|  128 |  1328 | 11 |
+|  256 |  2951 | 12 |
+|  512 |  6483 | 13 |
+| 1024 | 14116 | 14 |
+| 2048 | 30509 | 15 |
+
+The worst case constructible message in a square to have the biggest impact from switching from O(log(n)) to O(sqrt(n)) with n being the size of the message, has r rows in a square size of k. If r is larger than k/4 we need to take k/4 as the number of rows instead because of the first point in this message. With adopting this rule the messages look as follows:
+
+|   k  |   n   |  r |
+|:----:|:-----:|:--:|
+|    2 |     **2** |  **1** |
+|    4 |     **4** |  **1** |
+|    8 |    **16** |  **2** |
+|   16 |    **64** |  **4** |
+|   32 |   256 |  8 |
+|   64 |   589 | 10 |
+|  128 |  1328 | 11 |
+|  256 |  2951 | 12 |
+|  512 |  6483 | 13 |
+| 1024 | 14116 | 14 |
+| 2048 | 30509 | 15 |
+
+ Reminder: We did this calculation because we need O(log(n)) rows.
+
+## 3. Why can we not use the same trick that we used in Question 1 in a single row for more efficient proofs over the row roots?
+
+The node needs to be part of the commitment generation **and** part of the Merkle tree to the DataRoot for the trick to work. The diagram shows a Celestia square that is erasure coded and those parity shares are marked in green.
+H12 is part of the commitment generation and part of the Merkle tree to the DataRoot.
+It is only generated in the bigger square and not in the smaller square because in the smaller square you have to take into account the nodes over the parity shares.
+As H12 only exists in the bigger square the more efficient proofs only work in those squares.
```

Comment:
[question] It seems like the bottom most square in `./assets/rowroots-might-not-be-subtreeroot.png` also needs to be erasure coded so I don't follow why erasure coding the small square breaks the trick but erasure coding the larger square doesn't break the trick.

Reply:
I changed the picture so it should be clearer :) H12 does not exist in the smaller square but in the larger square it does.

Reply:
Ahh I see, thanks for updating the diagram.

Small correction in the bottom square: `ROW3` => `ROW2` and `ROW2` => `ROW3`